service/controller/resource/ipam/create.go

```diff
@@ -66,7 +66,17 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		r.logger.LogCtx(ctx, "level", "debug", "message", "finding allocated subnets")
 
 		allocatedSubnets, err = r.collector.Collect(ctx, obj)
-		if err != nil {
+		if IsNetworkRangeStillNotKnown(err) {
```

Comment:
I don't want to block this PR, but wouldn't make sense to move the comment to the `IsNetworkRangeStillNotKnown()` function so that we don't have to duplicate it?

Reply:
Updated comments a bit, so that:
- comments in `EnsureCreated` are about how the errors affects the reconciliation
- comments in `error.go` are about the error itself

P.S. The doubling was on purpose, so you see what's happening more quickly/easily, but now it's clear as well I believe :)